server/UserRegistration.java

```diff
@@ -0,0 +1,467 @@
+            updateuserdb.setString(6, userNewPass);
+            updateuserdb.setInt(7, userNewBal);
+            updateuserdb.setString(8, userNewSecretQuestion);
+
+
+            updateuserdb.executeUpdate();
+            conn.commit();
+            // The db will now push the output.
+
+
+        } catch (SQLException e) {
+            System.out.println(e.getMessage());
+        } finally {
+            try {
+                if (conn != null) {
+                    conn.close();
+                }
+            } catch (SQLException ex) {
+                System.out.println(ex.getMessage());
+            }
+        }
+
+
+        return null;
+    }
+
+
+    private static class newUserName {
+
+        Scanner sc = new Scanner(System.in);
+        String newName;
+
+        public String GetNameFromUser() {
+
+            System.out.println(" ");
+            System.out.print("                                               Welcome to NewBank Registration");
+            System.out.println(" ");
+            System.out.println(" ");
+
+            System.out.print("\nPlease enter your first name: ");
+            String newFirstName = sc.nextLine().toUpperCase();
+
+            System.out.print("\nPlease enter your last name: ");
+            String newLastName = sc.nextLine().toUpperCase();
+            newName = newFirstName + " " + newLastName;
+            System.out.print("\nThe name associated with this account will be:" + " " + newName);
+            String response;
+
+            do {
+                System.out.print("\nIs this correct (Y/N)?: ");
+                response = sc.nextLine().toUpperCase();
+                if (response.equals("N")) {
+                    System.out.print("Ok, you would like to change the name for the account");
+                    System.out.print("\nWhat will the new first name be: ");
+                    newFirstName = sc.nextLine().toUpperCase();
+                    System.out.println("\nWhat will the new last name be: ");
+                    newLastName = sc.nextLine().toUpperCase();
+                    newName = newFirstName + " " + newLastName;
+                    System.out.print("\nThe name associated with this account will be:" + " " + newName);
+                    System.out.print("\nIs this correct (Y/N)?: ");
+                    response = sc.nextLine().toUpperCase();
+                }
+            } while (!response.equals("Y"));
+            return newName;
+        }
+
+    }
+
+
+    private static class DateChecks {
+
+        Scanner sc = new Scanner(System.in);
+
+        boolean IsItDate = false;
+        String EnteredDate;
+        String userdateresponse;
+
+        public String GetDateFromUser() {
+
+            do {
+                System.out.print("\nFor your date of birth please enter in the format of (DD/MM/YYYY) including /");
+                System.out.println("\nDate of Birth: ");
+                String EnteredDate = sc.nextLine();
```

Comment:
This will make the date `null`, as you're re-declaring `EnteredDate` inside of this scope, so just remove the `String` here